src/features/common.rs

```diff
@@ -67,4 +53,159 @@ pub struct Options {
+
+    impl Drop for Runtime<'_, '_> {
+        fn drop(&mut self) {
+            use std::mem::drop;
+            unsafe {
+                drop(Box::from_raw(self.scope));
+                drop(Box::from_raw(self.context));
+                drop(Box::from_raw(self.handle));
+                drop(Box::from_raw(self.isolate));
+            };
+        }
+    }
+
+    #[allow(unused)]
+    impl<'s, 'i> Runtime<'s, 'i> where 's: 'i, {
+        pub(in crate::features) fn new() -> Self {
+            #[cfg(not(feature = "external-v8"))]
+            static V8_INIT: Once = Once::new();
+
+            #[cfg(not(feature = "external-v8"))]
+            V8_INIT.call_once(|| {
+                let platform = v8::new_default_platform(0, false).make_shared();
+                v8::V8::initialize_platform(platform);
+                v8::V8::initialize();
+            });
+
+            let isolate = Box::into_raw(Box::new(v8::Isolate::new(v8::CreateParams::default())));
+            let handle = unsafe { Box::into_raw(Box::new(v8::HandleScope::new(&mut *isolate))) };
+            let context = unsafe { Box::into_raw(Box::new(v8::Context::new(&mut *handle))) };
+            let scope = unsafe { Box::into_raw(Box::new(v8::ContextScope::new(&mut *handle, *context))) };
+
+            return Runtime {isolate, handle, context, scope};
+        }
+
+        pub(in crate::features) fn get_isolate(&self) -> &mut v8::OwnedIsolate {
+            return unsafe {&mut *self.isolate};
+        }
+
+        pub(in crate::features) fn get_handle(&self) -> &mut v8::HandleScope<'s,()> {
+            return unsafe {&mut *self.handle};
+        }
+
+        pub(in crate::features) fn get_context(&self) -> &mut v8::Local<'s, v8::Context> {
+            return unsafe {&mut *self.context};
+        }
+
+        pub(in crate::features) fn get_scope(&self) -> &mut v8::ContextScope<'i, v8::HandleScope<'s,v8::Context>> {
+            return unsafe {&mut *self.scope};
+        }
+
+        pub(in crate::features) fn run<S: AsRef<str>>(&self, s: S) -> Option<v8::Local<'s, v8::Value>> {
+            let scope = self.get_scope();
+            let script = v8::String::new(scope, s.as_ref())?.into();
+            let script = v8::Script::compile(scope, script, None)?;
+            return script.run(scope);
+        }
+    }
+
+    pub(in crate::features) trait UsingRuntime<'x,'s,'i> {
+        fn using_runtime<F, R>(&'x self, f: F) -> R where F: FnOnce(&Runtime<'s,'i>) -> R;
+    }
+
+    impl <'x,'s,'i> UsingRuntime<'x,'s,'i> for Runtime<'s,'i> where 's: 'i {
+        fn using_runtime<F, R>(&'x self, f: F) -> R where F: FnOnce(&Runtime<'s,'i>) -> R {
+            f(self)
+        }
+    }
+
+    impl UsingRuntime<'static,'static, 'static> for std::thread::LocalKey<Runtime<'static,'static>> {
+        fn using_runtime<F, R>(&'static self, f: F) -> R where F: FnOnce(&Runtime<'static,'static>) -> R {
+            self.with(f)
+        }
+    }
+}
+
+#[macro_export]
+macro_rules! with_v8 {
```

Comment:
A macro to help with using `Runtime`s and v8.

Usage:
```rs
let x = with_v8! {
    use <name> = <expr>;
    //code...
}
```
`<expr>` is an expression that gives an implementation of `UsingRuntime`, e.g. `Runtime::new()` or a reference to a `thread_local!` Runtime (LocalKey<Runtime>); `<name>` is a name to bind the runtime to for later usage. This line is required. The following code may use the runtime as well as macros `v8_str!(<expr>)`, `v8_bool!(<expr>)`, `v8_set!(<target>.<prop> = <value>)`, `v8_get!(<target>.<prop>`), and `v8_object!({ <prop>: <value>, ...})`. More helpers may be added in the future as needed. `with_v8!` also requires MTSC use the nightly compiler.